### PR TITLE
Basic logic server

### PR DESCRIPTION
**Pull request by SharonEizner**
logic in server

### CODE REVIEW

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import lombok.Data;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.Scanner;
+
+@Data
+public class Pawn {
+    private int id;
+    private String color;
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java:13]: Color should not be in server side.
Only in client side.

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+                        cur.getX(),
+                        cur.getY() - 2);
+                nextToVisit.add(Position);
+            }
+            board[cur.getY()][cur.getX()] = Cell.CHECKED;
+        }
+        return true;
+    }
+
+    public Cell[][] addWallToBoard(Wall wall, Cell[][] board) {
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:163]: Invalid board argument.
Should return true / false

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+    private void initBoard() {
+        for (int height = 0; height < this.board.length; height++) {
+            for (int width = 0; width < this.board[0].length; width++) {
+                this.board[height][width] = Cell.EMPTY;
+            }
+        }
+    }
+
+    private boolean checkPaths(Wall wall, Position[] positions, int[] heightsWin) {
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:89]: Thing about this arguments and function location (if this should be in board)

Comment (Guyeise1) [reply quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:89]: arguments should be player

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    public boolean addWall(Wall wall, Position[] positions, int[] heightsWin) {
+        if (this.checkAddingWall(wall, positions, heightsWin)) {
+            this.board = this.addWallToBoard(wall, this.board);
+            return true;
+        }
+
+        return false;
+    }
+
+    public boolean checkAddingWall(Wall wall, Position[] positions, int[] heightsWin) {
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:50]: same as checkPaths

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/enums/Cell.java [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.rl.mpquoridor.enums;
+
+public enum Cell {
+    EMPTY,
+    WALL,
+    PAWN,
+    CHECKED
```
Comment (Guyeise1) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/enums/Cell.java:7]: Should not be here

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/enums/Direction.java [hunk 6]
```diff
@@ -0,0 +1,8 @@
+package com.rl.mpquoridor.enums;
+
+public enum Direction {
+    Right,
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/enums/Direction.java:4]: enum value in java should be always CAPITAL ===> Direcrtion.RIGHT

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        Direction direction = wall.getDirection();
+        Position position = wall.getPosition();
+        int newX = position.getX() * 2 + 1;
+        int newY = position.getY() * 2 + 1;
+
+        if (direction == Direction.Right) {
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:56]: with enums the best practice is to use ".equals" and not == (also in strings and other objects)

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+
+        for (int i = 0; i < positions.length; i++) {
+            if (this.checkingExistingPath(newBoard, positions[i], heightsWin[i])) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    //BFS
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:105]: I'm sure that there is existing library which implements BFS
mayby there is something easy to use

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java (hunk 2 above)
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:163]: why the it gets and return the board??? (the board is a private member with getters and setters...)

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java [hunk 9]
```diff
@@ -0,0 +1,111 @@
...
+    private String name;
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java:14]: I think color and name should be under another pojo called Player

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            }
+        }
+
+    }
+
+    public int numOfWallsRemaining() {
+        return numOfWallsStarting - numOfWallsInUse;
+    }
+
+    public Wall[] addWall(Wall wall) {
```
Comment (shvshv44) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java:88]: should be under player
pawn has to be something on the board
player have a pawn

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java (hunk 1 above)
Comment (RavivGuy) [reply quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Pawn.java:13]: Like guy said

File: quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+    }
+
+    public boolean movePawn(Position fromPosition, Position toPosition) {
+        //TODO: check if this is a valid new position
+        this.board[fromPosition.getY() * 2][fromPosition.getX() * 2] = Cell.EMPTY;
+        this.board[toPosition.getY() * 2][toPosition.getX() * 2] = Cell.PAWN;
+        return true;
+    }
+
+    public boolean addWall(Wall wall, Position[] positions, int[] heightsWin) {
```
Comment (RavivGuy) [inline quoridor-logic-server/src/main/java/com/rl/mpquoridor/models/Board.java:41]: What is the meaningOf heightsWin?

**Pull request closed without merging**